conbench/app/index.py

```diff
@@ -1,25 +1,76 @@
-import flask as f
+from collections import defaultdict
+from dataclasses import dataclass
+
+from sqlalchemy import select
 
 from ..app import rule
 from ..app._endpoint import AppEndpoint, authorize_or_terminate
 from ..app.benchmarks import RunMixin
 from ..config import Config
+from ..db import Session
+from ..entities.run import Run
 
 
 class Index(AppEndpoint, RunMixin):
-    def page(self, runs):
+    def page(self, repo_runs_map):
         return self.render_template(
             "index.html",
             application=Config.APPLICATION_NAME,
             title="Home",
-            runs=runs,
-            search_value=f.request.args.get("search"),
+            repo_runs_map=repo_runs_map,
+            # Note(JP): search_value is not consumed in template
+            # search_value=f.request.args.get("search"),
         )
 
     @authorize_or_terminate
     def get(self):
-        runs = self.get_display_runs()
-        return self.page(runs)
+        # Following
+        # https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#selecting-orm-entities
+        runs = Session.scalars(
+            select(Run).order_by(Run.timestamp.desc()).limit(1000)
+        ).all()
+
+        # Note(JP): group runs by associated commit.repository value Note that
+        # consistency between benchmark results in the run is currently not
+        # granted: https://github.com/conbench/conbench/issues/864
+
+        repo_runs_map = defaultdict(list)
+
+        for r in runs:
+            rd = RunForDisplay(
+                ctime_for_table=r.timestamp.strftime("%Y-%m-%d %H:%M:%S UTC"),
+                commit_message_short=short_commit_msg(r.commit.message),
+                run=r,
```

Comment:
this is still experimentation in the WIP state of this PR. Basically I think we want to pass the raw `Run` object into the template, but in addition to that we also need some variables tailored for displaying in HTML. The latter is what the "augment" family of functions did before. I'd love to do this closer to where the HTML template is rendered.